File: python/linked-list-random-node.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
import random

class Solution:

    def __init__(self, head: Optional[ListNode]):
        self.data = []
        while head:
            self.data.append(head.val)
            head = head.next
        self.length = len(self.data)
        
    def getRandom(self) -> int:
        return self.data[random.randint(0, self.length-1)]

File: go/linked_list_random_node.go
package leetcode

import (
	"math/rand"
)

type ListNode struct {
	Val  int
	Next *ListNode
}
type Solution struct {
	Data   []int
	Length int
}

func Constructor(head *ListNode) Solution {
	s := Solution{}
	for head != nil {
		s.Data = append(s.Data, head.Val)
		head = head.Next
	}
	s.Length = len(s.Data)
	return s
}

func (this *Solution) GetRandom() int {
	return this.Data[rand.Intn(this.Length)]
}
